### ISSUE TITLE
Running host-updates failes with AttributeError traceback

### ISSUE THREAD

**Issue by radiowavesurfer** [furlongm/patchman#521]
Hi. I'm sometimes finding the host-updates (and also host-updates-alt) process failing like so:

```
Traceback (most recent call last):
  File "/usr/bin/patchman", line 686, in <module>
    main()
  File "/usr/bin/patchman", line 680, in main
    showhelp = process_args(args)
  File "/usr/bin/patchman", line 650, in process_args
    host_updates(args.host)
  File "/usr/bin/patchman", line 331, in host_updates
    host.find_updates()
  File "/usr/lib/python3/dist-packages/hosts/models.py", line 192, in find_updates
    update_ids = self.find_host_repo_updates(host_packages,
  File "/usr/lib/python3/dist-packages/hosts/models.py", line 235, in find_host_repo_updates
    pu_priority = pu_best_repo.priority
AttributeError: 'NoneType' object has no attribute 'priority'
```

I haven't fathomed exactly what combination of package state and repo state is causing this, but it looks like it's related to Ubuntu hosts where the Ubuntu advantage tools have "helpfully" enabled ESM (i.e. extended support lifetime) repos, but where I don't have an ESM subscription to be able to use them.

Looking at the code where the error is being thrown:
(hosts/models.py, from line 216)
```
for package in host_packages:
    highest_package = package
    best_repo = find_best_repo(package, hostrepos)
    priority = None
    if best_repo is not None:
        priority = best_repo.priority

    # find the packages that are potential updates
    pu_q = Q(name=package.name, arch=package.arch,
             packagetype=package.packagetype)
    potential_updates = repo_packages.filter(pu_q)
    for pu in potential_updates:
        if highest_package.compare_version(pu) == -1 \
                and package.compare_version(pu) == -1:

            if priority is not None:
                # proceed only if the package is from a repo with a
                # priority and that priority is >= the repo priority
                pu_best_repo = find_best_repo(pu, hostrepos)
                pu_priority = pu_best_repo.priority
                if pu_priority >= priority:
                    highest_package = pu
            else:
                highest_package = pu
```

Based on some extra warning messages I put into the code while watching it run, it looks like the function `find_best_repo()` is answering some value when called with the `package` variable (near the top of the code excerpt), and hence the `priority` variable is being set to some value (i.e. not `None`).

This then means that we go down the `if priority is not None:` branch, where `find_best_repo()` is called again but with the `pu` variable, and this time the function returns `None`, causing the attribute error when we then try to get the priority of the returned value.

In other words, the code assumes that if we've found a best repo for the package, we'll definitely also find one for the potential update, and doesn't guard against failing to find one.

Whether there is an actually correct course of action to take with regard to this odd situation with the ESM repos, I'm not sure, but I think it would be a good idea to at least avoid the error and traceback.

Thanks.

**Comment by furlongm** [comment]
Could you add what the repos look like in the report for one of these hosts?

Running `patchman-client` with `-d` should dump a couple of files with the repos in it.

**Comment by furlongm** [comment]
ESM repos seem fine in terms of priority etc when I tested them:

```
'deb' 'Ubuntu 18.04 x86_64 repo at https://esm.ubuntu.com/infra/ubuntu/dists/bionic-infra-updates/main/binary-amd64' '500' 'https://esm.ubuntu.com/infra/ubuntu/dists/bionic-infra-updates/main/binary-amd64'
'deb' 'Ubuntu 18.04 x86_64 repo at https://esm.ubuntu.com/infra/ubuntu/dists/bionic-infra-security/main/binary-amd64' '500' 'https://esm.ubuntu.com/infra/ubuntu/dists/bionic-infra-security/main/binary-amd64'
'deb' 'Ubuntu 18.04 x86_64 repo at https://esm.ubuntu.com/apps/ubuntu/dists/bionic-apps-updates/main/binary-amd64' '500' 'https://esm.ubuntu.com/apps/ubuntu/dists/bionic-apps-updates/main/binary-amd64'
'deb' 'Ubuntu 18.04 x86_64 repo at https://esm.ubuntu.com/apps/ubuntu/dists/bionic-apps-security/main/binary-amd64' '500' 'https://esm.ubuntu.com/apps/ubuntu/dists/bionic-apps-security/main/binary-amd64'
'deb' 'Ubuntu 18.04 x86_64 repo at http://security.ubuntu.com/ubuntu/dists/bionic-security/multiverse/binary-amd64' '500' 'http://security.ubuntu.com/ubuntu/dists/bionic-security/multiverse/binary-amd64'
'deb' 'Ubuntu 18.04 x86_64 repo at http://security.ubuntu.com/ubuntu/dists/bionic-security/universe/binary-amd64' '500' 'http://security.ubuntu.com/ubuntu/dists/bionic-security/universe/binary-amd64'
'deb' 'Ubuntu 18.04 x86_64 repo at http://security.ubuntu.com/ubuntu/dists/bionic-security/restricted/binary-amd64' '500' 'http://security.ubuntu.com/ubuntu/dists/bionic-security/restricted/binary-amd64'
'deb' 'Ubuntu 18.04 x86_64 repo at http://security.ubuntu.com/ubuntu/dists/bionic-security/main/binary-amd64' '500' 'http://security.ubuntu.com/ubuntu/dists/bionic-security/main/binary-amd64'
'deb' 'Ubuntu 18.04 x86_64 repo at http://archive.ubuntu.com/ubuntu/dists/bionic-backports/universe/binary-amd64' '100' 'http://archive.ubuntu.com/ubuntu/dists/bionic-backports/universe/binary-amd64'
'deb' 'Ubuntu 18.04 x86_64 repo at http://archive.ubuntu.com/ubuntu/dists/bionic-backports/main/binary-amd64' '100' 'http://archive.ubuntu.com/ubuntu/dists/bionic-backports/main/binary-amd64'
'deb' 'Ubuntu 18.04 x86_64 repo at http://archive.ubuntu.com/ubuntu/dists/bionic-updates/multiverse/binary-amd64' '500' 'http://archive.ubuntu.com/ubuntu/dists/bionic-updates/multiverse/binary-amd64'
'deb' 'Ubuntu 18.04 x86_64 repo at http://archive.ubuntu.com/ubuntu/dists/bionic-updates/universe/binary-amd64' '500' 'http://archive.ubuntu.com/ubuntu/dists/bionic-updates/universe/binary-amd64'
'deb' 'Ubuntu 18.04 x86_64 repo at http://archive.ubuntu.com/ubuntu/dists/bionic-updates/restricted/binary-amd64' '500' 'http://archive.ubuntu.com/ubuntu/dists/bionic-updates/restricted/binary-amd64'
'deb' 'Ubuntu 18.04 x86_64 repo at http://archive.ubuntu.com/ubuntu/dists/bionic-updates/main/binary-amd64' '500' 'http://archive.ubuntu.com/ubuntu/dists/bionic-updates/main/binary-amd64'
'deb' 'Ubuntu 18.04 x86_64 repo at http://archive.ubuntu.com/ubuntu/dists/bionic/multiverse/binary-amd64' '500' 'http://archive.ubuntu.com/ubuntu/dists/bionic/multiverse/binary-amd64'
'deb' 'Ubuntu 18.04 x86_64 repo at http://archive.ubuntu.com/ubuntu/dists/bionic/universe/binary-amd64' '500' 'http://archive.ubuntu.com/ubuntu/dists/bionic/universe/binary-amd64'
'deb' 'Ubuntu 18.04 x86_64 repo at http://archive.ubuntu.com/ubuntu/dists/bionic/restricted/binary-amd64' '500' 'http://archive.ubuntu.com/ubuntu/dists/bionic/restricted/binary-amd64'
'deb' 'Ubuntu 18.04 x86_64 repo at http://archive.ubuntu.com/ubuntu/dists/bionic/main/binary-amd64' '500' 'http://archive.ubuntu.com/ubuntu/dists/bionic/main/binary-amd64'
```

So I just created a PR to fix the issue - can you let me know if that fixes it for you?
